### ISSUE TITLE
[8.0.100-preview.4.23211.1] Anchor tag behavior changed when setting herf="#" in blazor app

### ISSUE THREAD

**Issue by Junjun-zhao** [dotnet/aspnetcore#47682]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

In dotnet-sdk-8.0.100-preview.4.23211.1, when <a> set herf="#" in blazor page, if user click the <a> link, it will stay on current page. But in previous version, it will navigate to main page.

**Application Name**: Blazorise
**OS**: Windows 10 21H2
**CPU**: X64
**.NET Build Number:** dotnet-sdk-8.0.100-preview.4.23211.1
**App Source checking at:**  https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1799823
**Or Github lin**k:
 https://github.com/Megabit/Blazorise

### Expected Behavior

It navigates to Home page, the url is "https://localhost:5000/#".

### Steps To Reproduce

**Repro steps:**

1.Create a default Blazor Server app with Visual Studio.
2. Add ```"<a href="#">Test Link</a>" ```on Counter.razor.
3. Build the project.
4. Open BlazorApp1.runtimeconfig.json to change the runtime version to let the app run against  8.0.100-preview.4.23211.1:
```
    "frameworks": [
      {
        "name": "Microsoft.NETCore.App",
        "version": "8.0.0-preview.4.23210.1"
      },
      {
        "name": "Microsoft.AspNetCore.App",
        "version": "8.0.0-preview.4.23207.6"
      }
]
```
5. Launch the app.
6. Navigate to Counter page.
7. Click the "Test Link".

**Actual Result:**

It stays on current page, when click the link, the url is "[https://localhost:5000/counter#](https://localhost:7076/counter#)". When mouse hover to the link, it shows https://localhost:5000/#  .

### .NET Version

8.0.100-preview.4.23211.1

### Anything else?

**Findings**

- In .NET Framework web app, if we add ```<a href="#">Test Link</a>``` on the page, when click on it, it stays on current page.

- In 8.0.100-preview.4.23211.1, when mouse hover to the link, it shows https://localhost:5000/# But when click it, it goes to [https://localhost:5000/counter#](https://localhost:7076/counter#)

@dotnet-actwx-bot @dotnet/compat

**Comment by surayya-MS** [comment]
Thanks for the report @Junjun-zhao !

Staying on the page by clicking `<a href="#">Test Link</a>` is a new expected behavior that was introduced recently.

> when mouse hover to the link, it shows https://localhost:5000/# But when click it, it goes to [https://localhost:5000/counter#](https://localhost:7076/counter#)

However this is a bug. It should show https://localhost:5000/counter#

**Comment by Junjun-zhao** [comment]
Verified it and the issue is fixed on dotnet-sdk-8.0.100-preview.4.23253.2
